packages/core/echo/echo-pipeline/src/dbhost/snapshot-manager.ts

```diff
@@ -57,13 +58,13 @@ export class SnapshotManager {
   }
 
   @timed(10_000)
-  async load(id: string): Promise<SpaceSnapshot> {
-    const local = await this._snapshotStore.loadSnapshot(id);
+  async load(ctx: Context, id: string): Promise<SpaceSnapshot> {
+    const local = await cancelWithContext(ctx, this._snapshotStore.loadSnapshot(id));
     if (local) {
       return local;
     }
 
-    const remote = await this._objectSync.download(id);
+    const remote = await cancelWithContext(ctx, this._objectSync.download(id));
```

Comment:
this is already good, but can you drill the cancellation deeper into objectSync (and add a unit test)

<img width="907" alt="image" src="https://github.com/dxos/dxos/assets/35851437/3b0de874-defa-41bd-ad75-cb028936d528">


Download works by adding a CID to the "want list" and then waiting for other peers to send it to you. so the cancellation would remove this from the "want list". Becareful that there might be multiple concurent requests to `download` with the same CID so you need to count them and remove the CID from the want list only if there's 0 active downloads for it.